core/src/main/java/org/apache/accumulo/core/fate/FateExecutor.java

```diff
@@ -201,17 +208,23 @@ protected Set<Fate.FateOperation> getFateOps() {
     return fateOps;
   }
 
+  public FateExecutorMetrics<T> getFateExecutorMetrics() {
+    return fateExecutorMetrics;
+  }
+
   /**
    * Initiates the shutdown of this FateExecutor. This means the pool executing TransactionRunners
    * will no longer accept new TransactionRunners, the currently running TransactionRunners will
-   * terminate after they are done with their current transaction, if applicable, and the work
-   * finder is interrupted. {@link #isShutdown()} returns true after this is called.
+   * terminate after they are done with their current transaction, if applicable, the work finder is
+   * interrupted, and the metrics created for this FateExecutor are removed from the registry (if
+   * metrics were enabled). {@link #isShutdown()} returns true after this is called.
    */
   protected void initiateShutdown() {
     transactionExecutor.shutdown();
     synchronized (runningTxRunners) {
       runningTxRunners.forEach(TransactionRunner::flagStop);
     }
+    fateExecutorMetrics.clearMetrics();
     workFinder.interrupt();
   }
```

Comment:
This will remove metrics from the registry when shutdown is initiated, not necessarily when the threads are done with the fate operations they are working on. Alternatively, could clear the metrics after all threads have terminated